oauth2client/contrib/gce.py

```diff
@@ -16,60 +16,113 @@
+To learn more go to: https://cloud.google.com/compute/docs/authentication
 """
+_SCOPES_ERROR = """\
+You have requested explicit scopes to be used with a GCE service account
+which are not available on the credentials. The scopes are set at VM instance
+creation time and can't be overridden in the request.
+To learn more go to: https://cloud.google.com/compute/docs/authentication
+"""
+_NOW = datetime.datetime.now
+
 
+class MetadataServerHttpError(Exception):
+    """Error for Http failures originating from the Metadata Server"""
 
-def _get_service_account_email(http_request=None):
-    """Get the GCE service account email from the current environment.
+
+def _get_metadata(http_request=None, path=None, recursive=True):
+    """Gets a JSON object from the specified path on the Metadata Server
 
     Args:
-        http_request: callable, (Optional) a callable that matches the method
-                      signature of httplib2.Http.request, used to make
-                      the request to the metadata service.
+        http_request: an httplib2.Http().request object or equivalent
+            with which to make the call to the metadata server
+        path: a list of strings denoting the metadata server request
+            path.
 
     Returns:
-        tuple, A pair where the first entry is an optional response (from a
-        failed request) and the second is service account email found (as
-        a string).
+        A deserialized JSON object representing the data returned
+        from the metadata server
     """
-    if http_request is None:
+
+    if not path:
+        path = []
+
+    if not http_request:
         http_request = httplib2.Http().request
+
+    r_string = '/?recursive=true' if recursive else ''
+    full_path = _METADATA_ROOT + '/'.join(path) + r_string
     response, content = http_request(
-        _DEFAULT_EMAIL_METADATA, headers={'Metadata-Flavor': 'Google'})
+        full_path,
+        headers={'Metadata-Flavor': 'Google'}
+    )
     if response.status == http_client.OK:
-        content = _from_bytes(content)
-        return None, content
+        decoded = _from_bytes(content)
+        if response['content-type'] == 'application/json':
+            return json.loads(decoded)
+        else:
+            return decoded
     else:
-        return response, content
+        msg = (
+            'Failed to retrieve {path} from the Google Compute Engine'
+            'metadata service. Response:\n{error}'
+        ).format(path=full_path, error=response)
+        raise MetadataServerHttpError(msg)
+
+
+def _get_access_token(http_request, email):
+    """Get an access token for the specified email from the Metadata Server.
+
+    Args:
+        http_request: an httplib2.Http().request object or equivalent
+            with which to make the call to the metadata server
+        email: The service account email to request an access token with
+
+    Returns:
+        A tuple (accessToken, token expiry)
```

Comment:
Fully describe the type of the return value. What is the type of the returned access token? What is the type of the returned token expiry? Why is one written thisWay and the other written this way? Also add trailing punctuation.